### PR TITLE
Cranelift: Implement tail calls on riscv64

### PR DESCRIPTION
**Pull request by fitzgen**
Builds on top of https://github.com/bytecodealliance/wasmtime/pull/6723

### CODE REVIEW

File: cranelift/fuzzgen/src/function_generator.rs [hunk 1]
```diff
@@ -2045,12 +2045,16 @@ where
...
-                let supports_tail_calls = self.isa.triple().architecture == Architecture::X86_64;
-                // TODO: We currently require frame pointers for tail calls
-                let has_frame_pointers = self.isa.flags().preserve_frame_pointers();
 
-                if is_tail_caller && has_tail_callees && supports_tail_calls & has_frame_pointers {
+                let supports_tail_calls = match self.isa.triple().architecture {
+                    // TODO: x64 currently requires frame pointers for tail calls.
+                    Architecture::X86_64 => self.isa.flags().preserve_frame_pointers(),
+                    Architecture::Aarch64(target_lexicon::Aarch64Architecture::Aarch64) => true,
+                    // TODO: Other platforms do not support tail calls yet.
```
Comment (elliottt) [inline cranelift/fuzzgen/src/function_generator.rs:2053]: Should riscv64 also have tail calls enabled when fuzzing?

Comment (fitzgen) [reply cranelift/fuzzgen/src/function_generator.rs:2053]: Yes it should. Good catch, thanks!

Comment (afonso360) [reply cranelift/fuzzgen/src/function_generator.rs:2053]: I've been running this on the fuzzer, so far nothing has shown up yet.

**Review by elliottt — APPROVED** [review]
Looks great!

File: cranelift/filetests/filetests/runtests/return-call-loop.clif [hunk 2]
```diff
@@ -1,9 +1,10 @@
 test run
 set preserve_frame_pointers=true
 target x86_64
-;; target aarch64
-;; target aarch64 sign_return_address
-;; target aarch64 has_pauth sign_return_address
+target aarch64
+target aarch64 sign_return_address
+target aarch64 has_pauth sign_return_address
```
[the remark this answers is not in the document]
Comment (fitzgen) [reply cranelift/filetests/filetests/runtests/return-call-loop.clif:6]: No, I just forgot to enable the tests in that PR.

**Pull request merged**